app/scripts/modules/core/src/search/widgets/Filter.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { IFilterType } from './SearchFilterTypeRegistry';
+
+export interface IFilterProps {
+  filterType: IFilterType;
+  isActive: boolean;
+  onKeyUp(event: React.KeyboardEvent<HTMLElement>): void;
+}
+
+@autoBindMethods
+export class Filter extends React.Component<IFilterProps> {
+  constructor(props: IFilterProps) {
+    super(props);
+  }
+
+  private handleKeyUp(event: React.KeyboardEvent<HTMLElement>): void {
+    this.props.onKeyUp(event);
+  }
+
+  public render(): React.ReactElement<Filter> {
+
+    const { isActive } = this.props;
+    const className = classNames({
+      'filter': true,
+      'filter--focus': isActive,
+      'filter--blur': !isActive
+    });
+
+    const { modifier, text } = this.props.filterType;
+    return (
+      <div
+        role="option"
+        className={className}
+        onKeyUp={this.handleKeyUp}
+      >
+        <div className="filter__text">{text}</div>
+        <div className="filter__modifier">[{modifier.toLocaleUpperCase()}:]</div>
```

Comment:
According to our [conventions document](https://github.com/spinnaker/deck/wiki/Conventions), we use BEM for class names so I was just following that.  I do want to be consistent but I feel like I'm getting mixed messages when I'm told "it's in the coding conventions doc" and when I follow it I also get questions.  Is that not fair?  If the rule of thumb is "Follow the coding conventions document unless the code dictates otherwise," I'm not sure there's value in having coding conventions.  What say ye?